.changes/unreleased/Under the Hood-20221206-094015.yaml

```diff
@@ -0,0 +1,7 @@
+kind: Under the Hood
+body: Prevent doc gen workflow from running on forks
+time: 2022-12-06T09:40:15.301984-06:00
+custom:
+  Author: stu-k
+  Issue: "6386"
+  PR: "6390"
```

Comment:
This change doesn't need a changelog since it's not something actually released.